score-lang-content-verifier/src/main/java/org/openscore/lang/tools/verifier/VerifierHelper.java

```diff
@@ -0,0 +1,134 @@
+/*******************************************************************************
+ * (c) Copyright 2014 Hewlett-Packard Development Company, L.P.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License v2.0 which accompany this distribution.
+ *
+ * The Apache License is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *******************************************************************************/
+package org.openscore.lang.tools.verifier;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.Validate;
+import org.apache.log4j.Logger;
+import org.openscore.lang.compiler.SlangCompiler;
+import org.openscore.lang.compiler.modeller.model.Executable;
+import org.openscore.lang.compiler.modeller.model.SlangFileType;
+import org.openscore.lang.compiler.scorecompiler.ScoreCompiler;
+import org.openscore.lang.entities.CompilationArtifact;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import static org.openscore.lang.compiler.SlangSource.fromFile;
+
+/*
+ * Created by stoneo on 2/9/2015.
+ */
+
+/**
+ * Verifies all files with extensions: .sl, .sl.yaml or .sl.yml in a given directory are valid
+ */
+public class VerifierHelper {
+
+    @Autowired
+    private SlangCompiler slangCompiler;
+
+    @Autowired
+    private ScoreCompiler scoreCompiler;
+
+    private final static Logger log = Logger.getLogger(SlangFilesVerifier.class);
+
+    private String[] SLANG_FILE_EXTENSIONS = {"sl", "sl.yaml", "sl.yml"};
+
+    private Map<String, Executable> slangModels = new HashMap<>();
+
+    private Map<String, CompilationArtifact> compiledArtifacts = new HashMap<>();
```

Comment:
why do you hold it on the class state and don't return it from the method and pass to the next?